### DISCUSSION TITLE
how to train embeddings just on earrings and clothes

### DISCUSSION THREAD

**Discussion by runner22k** [d8ahazard/sd_dreambooth_extension#808] [Q&A] [answered]
Can anyone help me with creating embeddings. I want to train 3 separate embedding on 3 subjects. (or is it possible to train on all 3, that would be awesome) I don't know how to tell Automatic1111 to choose only the dress from the photo and not jewelry, choose just earrings and nothing else. 

1. Just earrings
2. Just the dress
3. just jewelleries 

when we call dress, it should apply only dress type, and similarly with jewelries and earrings.

**Comment by Ackerlight** [comment, marked as answer]
An embedding is basically an embedded txt2img macro.  It works completely on the txt2img layer and the description should be the opposite of how you normally create a description for a model (for example). 

From personal experience, you may need to break this up into multiple embeddings.  If you are all right with that, then describe every detail in the image **except** what you want.  The embedding needs to infer (learn from what is not being described) what is missing from the images and it will recreate what is missing using txt2img.

Examples
1. A close up photo of a woman's ear.  (AI will focus on the ears and see you did not describe the shiny things hanging off them.  So will learn to recreate earrings from what you did not describe).
2. A woman with long blond hair standing in a room with hardwood floor, white walls and white curtains and wearing high heels.  (AI will see the woman and notice you did not describe what clothing she is wearing.  So may begin to look at what is similar between photos and notice her dress).
3. A close up photo of a woman's neck.  (AI may notice here necklace). 
   A close up photo of a woman's wrist.  (AI will look at her wrist and see you did not describe the watch or bracelet.  So will begin to look for similar bracelet and recreate it).

Now if you instead say "A woman wearing a blue sapphire princess cut necklace on her neck" the txt2img macro will ignore the necklace and never search or learn about it.  Because you already included it in the txt2img prompt.  Anything you include in the prompt will not be included in the embedding and the embedding will assume the user will need to type that every time as part of the text.  It will be completely ignored for training.

**Reply by elsucht** [reply]
What do you suggest for instance prompt and class prompt in case of train such tiny/detailed objects?

**Reply by Ackerlight** [reply]
Exactly what I wrote above.

Keep them very, very general prompts.  This will make the AI focus on the area, but not the details.  It will find the details and other objects itself.  Only describe in general terms.  Any extra specific details in the txt2img prompt WILL NOT be in the embedding.  Embeddings only create a macro to make what is NOT being described.  The only one that may be specific may be the one for the clothing.  Since you can describe the woman and her appearance and what is in the room.  As that is NOT important and should be cut out of the embedding.  Important to have similar images so that the AI can infer what IS important between multiple images.  So do not use the same woman for example or the AI may try and recreate that exact woman's appearance instead of just the clothing.

For a ring:  
Close up photo of a woman's hand

For a necklace:  
Close up photo of a woman's neck

For a dress/clothing:  
Photo of a woman standing in a room

For earrings:  
Close up photo of a woman's ear

For a bracelet:  
Close up photo of a woman's wrist

For a person:  (for this one it may be best to just use no prompt at all.  Just use a folder of images and give it a general class).
A man
A woman

**Reply by elsucht** [reply]
would you describe human features visible in the image like skin color, hair color,etc. too?

**Reply by Ackerlight** [reply]
Only if you do not want those features to be captured in the embedding. Anything you directly state in the txt2img prompt or keyword file will be skipped.  It only infers/finds what is not listed and unknown.  It is exactly opposite from prompting a model or LoRA.  Only list what you want it to ignore and skip.  Do not list what is important.

Remember an embedding only works on the txt2img prompt level.  If it is already  part of the prompt then SD will consider it already known and will skip/remove it from the macro it is trying to create.  So only list things you do not want or they will be removed from the embedded macro.

**Reply by Ackerlight** [reply]
As some people are learning, you can replace a 75 token maxed txt2img prompt with a single 4 token embedding.  (You are starting to see this with negative embeddings).  Embeddings strictly work on the txt2img prompt layer to recreate complex prompts.

Remember this is just a text macro.  If you can create something using a complex txt2img prompt and it is already known by your base model then an embedding can recreate it.  It cannot store new features or make any changes not already part of the base model.  It can sort of move pixels, face and body features around a bit so they resemble a source image.  But if the source is too complex or has body markings like specific tattoos, birthmarks, moles, etc... those cannot be captured in an embedding unless those are already in the base model.

In the example about jewelry - if the base model does not know what a princess cut diamond is already then an embedding will never be able to recreate it.  This is one example, but it cannot recreate anything the base model does not already know about.  For that you need to create a new model and train it using Dreambooth or EveryDream to add the new concept and add in new diffusion data.

**Reply by minienglish1** [reply]
That's great to know.  I've read several papers and blogs about embeddings multiple times and I still didn't realize it worked like that.

Where did you learn that?  Do you know where I can find an information source that covers other related concepts in such detail?

Thanks.

**Reply by Ackerlight** [reply]
Mostly from the SD Discord and other sources on the basics.  The rest is from personal experience having created about 80 or so embeddings and going through trial and error.

**Reply by minienglish1** [reply]
Thanks.

**Reply by andupotorac** [reply]
Have you created any embeddings of clothes you're willing to share so we can test how they work? Will they work on any character - be it a person, or an animal in human form for example?

**Reply by andupotorac** [reply]
Wanted to follow-up on this. Yes, they work on any character, and style it seems. I just trained a few clothes.

**A cap**
![image](https://github.com/d8ahazard/sd_dreambooth_extension/assets/3269738/3ddfbad2-a631-46cf-8e6f-4541bb66afce)
![image](https://github.com/d8ahazard/sd_dreambooth_extension/assets/3269738/0be652fd-debf-4649-a7b2-083fb273df7c)

**Comment by runner22k** [comment]
:) Thank you very much for a detailed description @Ackerlight . Sorry for delayed reply. I didn't had access to system from few days.
